codalab/codalab/settings/base.py

```diff
@@ -450,7 +450,7 @@ class Base(Configuration):
     # Keep celery from becoming unresponsive
     CELERY_ACKS_LATE = True
     CELERYD_PREFETCH_MULTIPLIER = 1
-    CELERYD_TASK_SOFT_TIME_LIMIT = 180  # 3 minutes
+    CELERYD_TASK_SOFT_TIME_LIMIT = 60 * 60 * 12  # 12 hours (create_storage_analytics_snapshot can take a lot of time the first time it runs)
```

Comment:
This may have unintended consequences for submission processing.

Would you be able to test what happens with a submission that just hangs up the worker? (IE: Sleeps for 15 minutes?)

Reply:
I found that we can apply task-specific time limit. I'll revert `CELERYD_TASK_SOFT_TIME_LIMIT` to what it was before and apply my new task its own longer time limit